docs/api.md

```diff
@@ -81,3 +80,4 @@ These are the API group versions which are currently available when using Antrea
 | CRD | CRD version | Introduced in | Deprecated in | Removed in |
 |---|---|---|---|---|
 | `ClusterGroup` | v1alpha2 | v1.0.0 | v1.1.0 | v2.0.0 |
+| `Tier` | v1alpha1 | v1.0.0 | v1.13.0 | v2.0.0 |
```

Comment:
We didn't have such a long time window between deprecation and removal for this one.
However, it was an Alpha API and we are upgrading to Antrea v2 so that's ok by me. I think we also agreed on this removal when we first started discussing Antrea v2 and bumping up Alpha APIs to Beta.

Adding @tnqn for his opinion.
My only concern would be a user upgrading from an older Antrea version (e.g. 1.11) to Antrea v2.0. In this case, the v1beta1 version of API didn't exist in v1.11, so `antctl upgrade api-storage` cannot be used. My assumption would be that direct upgrade to v2.0 would not be possible in this case, and that users would need to upgrade to an intermediate version (1.13, 1.14, or 1.15) first. The fact that the 2 API resources are technically the "same" (conversion strategy is None) should be irrelevant here.

I am also tagging @edwardbadboy.  We should make sure that the inability to _directly_ upgrade from 1.11, 1.12 to Antrea v2 will not be an issue before we can merge this.

If we do remove this API, I assume we will also remove `v1alpha3.ClusterGroup`?

Reply:
Thanks for tagging me here. I think it not an issue to upgrade to an intermediate version first, for example 1.11 -> 1.13, 1.11 -> 1.15, then upgrade to 2.0 or 2.1.

Reply:
We commit "Antrea upgrades to a new release up to 4 minor versions newer", but this is a major version so I assume it's fine. Before releasing v2.0, we should perhaps confirm which path is supported and document a best practice. It seems we could allow 1.13 and above upgrade to 2.0, which sounds not bad.

Reply:
Yes, we should document this before the v2 release

Reply:
I just found Tier is not the only case, now I wonder why we remove Tier alone and keep other resources of the same situation, including NetworkPolicy, ClusterNetworkPolicy, ClusterGroup and Group. Was Tier marked for removal in 2.0 by accident?

Reply:
My understanding (based on https://github.com/antrea-io/antrea/issues/4832) is that at least there was agreement to remove all deprecated Alpha APIs as part of Antrea v2 (along with all Beta APIs which were deprecated at least 9-12 months ago). Has this changed?
Not sure why Tier was treated differently in the doc, but I wouldn't read too much into it.

Reply:
Yes, we should. @luolanzone perhaps it would make more sense to do all of them at once, although I am also fine with an individual PR for each API?

Reply:
Sure, I can remove all v1alpha1 CRD in this PR. I forgot that we have an agreement to remove all v1alpha1 CRDs. This PR is to based on the discussion here https://github.com/antrea-io/antrea/pull/6089#discussion_r1542338375. I was trying to make the change consistent with api.md. I will remove all v1alpha1 and the api.md as well.

Reply:
Either approach works for me.